app/helpers/notifications_helper.rb

```diff
@@ -38,34 +36,34 @@ def notification_header(notification)
     text = if notification.medium?
       medium_notification_card_header(notifiable)
     elsif notification.course? || notification.lecture?
-      t('notifications.course_selection')
+      t("notifications.course_selection")
     elsif notification.lecture_announcement?
       announcement_notification_card_header(notifiable)
     else
-      link_to t('mampf_news.title'), news_path, class: 'text-dark'
+      link_to(t("mampf_news.title"), news_path, class: "text-dark")
     end
-    text.html_safe
+    text.html_safe # rubocop:todo Rails/OutputSafety
   end
 
   # provide text for body of notification card
   def notification_text(notification)
     notifiable = notification.notifiable
     text = if notification.medium?
-      t('notifications.new_medium')
+      t("notifications.new_medium")
     elsif notification.course?
       course_notification_card_text(notifiable)
     elsif notification.lecture?
       lecture_notification_card_text(notifiable)
     else
-      t('notifications.new_announcement')
+      t("notifications.new_announcement")
     end
-    text.html_safe
+    text.html_safe # rubocop:todo Rails/OutputSafety
   end
```

Comment:
Similar to the above, only this time `course_notification_card_text` and `lecture_notification_card_text` are constructed from the locales, but this time with parameters. It is unclear to me if it would be okay add an html_safe within both of these methods (and  to both `t('...')` in `notification_text`.)

Reply:
I think we shouldn't use `html_safe` here. For example, `lecture_notification_card_text` is constructed from `lecture.course.title`. This is a database field. In my opinion, we should construct the code such that user input is always sanitized *before* it goes to the db. This way, we don't have to worry and sanitize all the time when we query the database. In the case of `lecture_notification_card_text` we even assume that the db field is sanitized as we are using the `_html` method  which is automatically marked as HTML safe by Rails ([see here](https://guides.rubyonrails.org/i18n.html#using-safe-html-translations)).

However, there are lots of different opinions on this [here](https://stackoverflow.com/questions/1274558/when-to-filter-sanitize-data-before-database-insertion-or-before-display). And they have good arguments *not* to escape input and instead just filter it before it gets to the db. For example, when users can store HTML content and wand to edit it, then we shouldn't escape the tags...

What's the way we are doing this right now? Why do we use `html_safe` many times in different views, e.g.
`<%= chapter.details.html_safe %>` in `_chapter.html.erb`. Does that mean we have already sanitized the chapter details when the user submitted the form and we saved it to the db?

Edit: Ok, this [great post](https://stackoverflow.com/a/21886874/9655481) cleared things up to me. Briefly: as long as we use Active Record in the intended way, it's just fine and Rails does the job (as usual ;)). As a [comment over there](https://stackoverflow.com/questions/21886170/best-way-to-go-about-sanitizing-user-input-in-rails#comment136881268_21886874) I've provided some sources for the statements.

Reply:
see above (fixed in b26b11f2)